staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -263,7 +263,7 @@ func (d *DiscoveryClient) downloadLegacy() (
 	var resourcesByGV map[schema.GroupVersion]*metav1.APIResourceList
 	// Switch on content-type server responded with: aggregated or unaggregated.
 	switch {
-	case isV2Beta1ContentType(responseContentType):
+	case ContentTypeIsGVK(responseContentType, schema.GroupVersionKind{Group: "apidiscovery.k8s.io", Version: "v2beta1", Kind: "APIGroupDiscoveryList"}):
```

Comment:
pull the schema.GroupVersionKind{...} bit into a private package var for readability